### PR TITLE
fix: adds relative image url path for self hosters without s3

### PR DESCRIPTION
**Pull request by pandeymangg**
## What does this PR do?

For people self hosting formbricks and using the local storage (`uplaoads` directory) for file uploads, the next/image package throws errors about not being able to load the image because of an unrecognized url domain for the image. This happens because the next config is not able to add the domains for `WEB_APP_URL` at the runtime. 

This PR fixes the error for the local storage urls / images by transforming the local urls to use a relative path.

Fixes # (issue)

## How should this be tested?

- Upload any image without S3 config, avatar or welcome card for example.
- Image should be rendered with the next/image package

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: packages/lib/storage/service.ts [hunk 1]
```diff
@@ -188,6 +188,14 @@ export const getUploadSignedUrl = async (
   if (!IS_S3_CONFIGURED) {
     try {
       const { signature, timestamp, uuid } = generateLocalSignedUrl(updatedFileName, environmentId, fileType);
+      let fileUrl = "";
+
+      // self-hosted and no S3
+      if (!IS_FORMBRICKS_CLOUD) {
```
Comment (mattinannt) [inline packages/lib/storage/service.ts:194]: What if a self-hosted user is using S3?
Wouldn't `IS_S3_CONFIGURED` make more sense here?

**Review by mattinannt — COMMENTED** [review]
@pandeymangg thanks a lot for the PR 😊 Looks great, only one concern:

File: packages/lib/storage/service.ts (hunk 1 above)
Comment (pandeymangg) [reply packages/lib/storage/service.ts:194]: yeah this wouldn't work with users that are self hosting and have s3 configured, the next/image component does not load the image from S3 with a relative url 😞

**Comment by mattinannt** [comment]
@pandeymangg Please make the changes as discussed in the call 😊
- Always use absolute URL for file storage
- Use a remote-patterns wildcard if `WEBAPP_URL` is not set ( https://stackoverflow.com/questions/71235874/how-to-allow-all-domains-for-image-nextjs-config ), use the WEBAPP_URL hostname if set
- Only include the S3 remote patterns if S3 variables are present and build it dynamically from the variables

**Review by mattinannt — APPROVED** [review]
@pandeymangg thanks for making the changes 😊 Let's merge this 💪🚀

**Pull request merged**